src/Model/ModelManager.php

```diff
@@ -36,24 +36,32 @@
 use Sonata\DoctrineORMAdminBundle\Datagrid\OrderByToSelectWalker;
 use Sonata\DoctrineORMAdminBundle\Datagrid\ProxyQuery;
 use Sonata\Exporter\Source\DoctrineORMQuerySourceIterator;
-use Symfony\Component\Form\Exception\PropertyAccessDeniedException;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
 
 class ModelManager implements ModelManagerInterface, LockInterface
 {
     public const ID_SEPARATOR = '~';
+
     /**
      * @var ManagerRegistry
      */
     protected $registry;
 
+    /**
+     * @var PropertyAccessorInterface
+     */
+    protected $propertyAccessor;
+
     /**
      * @var EntityManager[]
      */
     protected $cache = [];
 
-    public function __construct(ManagerRegistry $registry)
+    public function __construct(ManagerRegistry $registry, ?PropertyAccessorInterface $propertyAccessor = null)
```

Comment:
I think we should change dependency injection to add this new param on constructor

Reply:
By the way, I added it because it was done like this in this PR https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/pull/368, but we don't always add the dependency in the constructor in SonataAdmin.

I have trouble to see a real reason to do this since there is a 
```
PropertyAccess::createPropertyAccessor()
```
method.

Do you know a good reason for this optional injection ?

Reply:
For me there could be some reasons:

1. To have consistency with the rest of injections.
2. To avoid executing code on constructors (ideally)
3. The injected property access is configured by https://github.com/symfony/symfony/blob/0ab5eeddf97d0a277c5ad956a51365aba93b27b9/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L886 But I dont know if that is a good reason or a bad one.

Do you know what configuration we get when creating our own instance?

How do we deal normally with this property accessor?

Reply:
> How do we deal normally with this property accessor?

It depends:
https://github.com/sonata-project/SonataAdminBundle/search?q=createpropertyaccessor&unscoped_q=createpropertyaccessor

Maybe should I add a deprecation if the dependency is not provided and remove the `= null` in the next major ? WDYT ?

Reply:
I think that would be great, unless there is a good reason to use the createPropertyAccessor I prefer to just inject the framework bundle one